### PR TITLE
DDP-7569. Lombok introduction

### PR DESCRIPTION
**Pull request by Trimax**
This PR doesn't contain any business logic changes (please verify that).
This is the first PR from the series (I hope) that will introduce wider usage of Lombok.
This PR contains changes to DTO classes. Instead of using vanilla java's getters, setters and constructors we use Lombok annotations that will generate all the stuff in preprocessing stage.

If you're using IntelliJ IDEA there is a Lombok plugin available. Please make sure you installed it.
Снимок экрана 2022-02-14 в 11 25 31

### CODE REVIEW

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmKitRequest.java [hunk 1]
```diff
@@ -1,79 +1,17 @@
...
-
-    /**
-     * The identifier for the kit request used by DSM as a handle.
-     * Check elsewhere for what we set this to.
-     *
-     * @return the kit request id
-     */
-    public String getKitRequestId() {
-        return kitRequestId;
-    }
```
Comment (MocanaAtBroad) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmKitRequest.java:47]: Some of these javadoc on the getters look useful and important.  Can we move the jdoc comments to the instance variable definitions?

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmStudyParticipant.java [hunk 2]
```diff
@@ -20,141 +22,6 @@
...
-    public void setLegacyShortId(String legacyShortId) {
-        this.legacyShortId = legacyShortId;
-    }
-
-    /**
-     * Report the validation status of the mailing address.
-     * “0” for an address that EasyPost says is invalid.
-     * “1” for an address that EasyPost says is valid and contains suggested EasyPost corrections.
-     * “2” for an address that EasyPost says is valid but does not contain EasyPost suggested corrections.
-     */
```
Comment (MocanaAtBroad) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmStudyParticipant.java:112]: This looks like use javadoc  stuff too.

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/housekeeping/BackupJobDto.java [hunk 3]
```diff
@@ -1,47 +1,25 @@
 package org.broadinstitute.ddp.db.dto.housekeeping;
 
+import lombok.AllArgsConstructor;
+import lombok.Data;
 import org.jdbi.v3.core.mapper.reflect.ColumnName;
 import org.jdbi.v3.core.mapper.reflect.JdbiConstructor;
 
+@Data
+@AllArgsConstructor(onConstructor = @__(@JdbiConstructor))
```
Comment (MocanaAtBroad) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/housekeeping/BackupJobDto.java:9]: Am I right that the generated constructor would have arguments in the same order as the instance variables are listed in the class definition (i.e., (String runName, long startTime, etcetc)?

**Review by MocanaAtBroad — COMMENTED** [review]
We using the "right" version of Lombok already?

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmKitRequest.java (hunk 1 above)
Comment (Trimax) [reply pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmKitRequest.java:47]: Yep, I'll move the docs. But I don't think that javadoc will work properly if you would like to generate it.

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/housekeeping/BackupJobDto.java (hunk 3 above)
Comment (Trimax) [reply pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/housekeeping/BackupJobDto.java:9]: Yes, you're right.

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmKitRequest.java (hunk 1 above)
Comment (MocanaAtBroad) [reply pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dto/dsm/DsmKitRequest.java:47]: We have never generated javadocs. More interested that comments are preserved when browsing the code.

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**